packages/jss-plugin-isolate/src/index.js

```diff
@@ -48,7 +47,7 @@ const ignoreParents = {
 const shouldIsolate = (rule: StyleRule, sheet, options) => {
   const {parent} = rule.options
 
-  if (parent && ignoreParents[parent.type]) {
+  if (parent && typeof parent.type === 'string' && ignoreParents[parent.type]) {
```

Comment:
would it work with a `type in parent` or just `parent.type` check? I wonder if string check is required

Reply:
parent.type doesn't work. Flow has a problem resolving the type for the `parent.type` and that's why the string check is necessary.

Reply:
type in parent too?

Reply:
The only place where type is not defined is when parent is a StyleSheet, in theory we could just typecast to a Rule, so this check won't be needed. Its a shame we have to add runtime checks for the sake of types

Reply:
We could also do this:

```js
  let {parent} = rule.options
  
  parent = ((parent: any): ConditionalRule | KeyframesRule)

  if (parent && ignoreParents[parent.type]) {
    return false
  }
```

Reply:
Yes, thats what I ment with casting, not sure how its called properly, I think its better, we basically manually tell the type system what it is, so it doesn't throw

Reply:
This works as well:
```js
  if (parent && (parent.type === 'keyframes' || parent.type === 'conditional')) {
    return false
  }
```

Reply:
types get removed from the runtime, these checks - don't

Reply:
Yeah but the above solution will result in `result = result`, or is it also being removed? When we check against the strings we don't need the type casting

Reply:
`parent = ((parent: any): ConditionalRule | KeyframesRule)` would becompletely removed afaik, whatever check you do in actual js code will not

Reply:
oh actually I am not sure if its gonna be removed after flow types are removed, but for sure after uglify

Reply:
Agree with you. That's why I would go for `parent.type === 'keyframes' || parent.type === 'conditional'` because this doesn't require the typecasting and we have two property access operations anyway.

Reply:
Huh, I thought actually that you would add with this 2 runtime checks, but with this its basically as it was before, just typechecked `result=result` check only in non-minified version

```
  parent = ((parent: any): ConditionalRule | KeyframesRule)

  if (parent && ignoreParents[parent.type]) {
```

Reply:
oh wait, you are going to replace with this ignoreParents[parent.type] checks ... yes thats not adding anything its 2 checks vs. 2 checks

Reply:
yeah it seems better